### PR TITLE
[socket] add missing header guard

### PR DESCRIPTION
**Pull request by wgtdkp**

### CODE REVIEW

File: src/library/socket.hpp [hunk 1]
```diff
@@ -31,6 +31,10 @@
  *   The file includes definitions of Socket.
  */
 
+
+#ifndef SOCKET_HPP_
```
Comment (bukepo) [inline src/library/socket.hpp:35]:
```suggestion
#ifndef OT_COMM_LIBRARY_SOCKET_HPP_
```

Comment (wgtdkp) [reply src/library/socket.hpp:35]: Since the header file is only accessible inside `src/library`, do we need the prefix here?

Comment (wgtdkp) [reply src/library/socket.hpp:35]: maybe `LIBRARY_*`

Comment (gocarlos) [reply src/library/socket.hpp:35]: I'd use `#pragma once` as this makes sure, no crashes with other macros and is supported by all compilers available for  CPP

Comment (gocarlos) [reply src/library/socket.hpp:35]: I'd use `#pragma once` as this makes sure, no crashes with other macros and is supported by all compilers available for  CPP

Comment (wgtdkp) [reply src/library/socket.hpp:35]: @gocarlos `#gragma once` doesn't seem to be a C++ standard.

Comment (gocarlos) [reply src/library/socket.hpp:35]: No, it is not, pragmas are vendor specific stuff.
Though is supported by all compilers this library supports and is used be a ton of c/cpp libs...
I don't have a strong opinion on this, i’m just pragmatic and saw many errors with other include guards

Comment (wgtdkp) [reply src/library/socket.hpp:34]: okay, I would prefer to keep current header guards to be consistent with openthread projects.

Comment (bukepo) [reply src/library/socket.hpp:34]: https://google.github.io/styleguide/cppguide.html#The__define_Guard

Comment (wgtdkp) [reply src/library/socket.hpp:34]: @bukepo Will do that in separate PR.

**Comment by wgtdkp** [comment]
@gocarlos @bukepo more comments?

**Pull request merged**